simulator/bpu/bpentry.h

```diff
@@ -73,13 +73,22 @@ class BPEntryOneBit final : public BPEntry
     /* update */
     void update( bool is_taken, Addr target)
     {
+
+        if ( is_taken && _target != target)
+        {
+            /* reset */
+            state = default_state;
+            _target = target;
+        } else 
+        {
+            state = static_cast<State>( is_taken);
+        }
```

Comment:
You duplicated code instead of merging. Please, use "then" part with "reset()" and append it with your "else" part.